ehcache-impl/src/main/java/org/ehcache/config/builders/ResourcePoolsBuilder.java

```diff
@@ -145,6 +146,9 @@ public ResourcePoolsBuilder with(ResourceType<SizedResourcePool> type, long size
    * @throws IllegalArgumentException if the set of resource pools already contains a heap resource
    */
   public ResourcePoolsBuilder heap(long size, ResourceUnit unit) {
+    if (unit instanceof MemoryUnit){
```

Comment:
As it stands this won't catch people using XML configuration unless the `ResourceConfigurationParser` is rewritten  to call the builder methods.

Reply:
I have removed this, Made single place change for both xml & java based configuration - `SizedResourcePoolImpl`